File: strings/lengthOfLongestSubstring.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        # brute
        ans=0

        for i in range(len(s)):
            hash_set = [0]*256

            for j in range(i, len(s)):
                if hash_set[ord(s[j])]==1:
                    break

                current_len=j-i+1

                hash_set[ord(s[j])]=1

                ans=max(ans, current_len)

        return ans

File: strings/lengthOfLongestSubstring.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        // brute
        int maxLen=0;

        for(int i=0; i<s.size(); i++){
            vector<int> hash(256, 0);

            for(int j=i; j<s.size(); j++){
                if(hash[s[j]]==1) break;
                hash[s[j]]=1;
                maxLen=max(maxLen, j-i+1);
            }
        }
        return maxLen;
    }
};